logs/kubelessLogs.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const _ = require('lodash');
+const Api = require('kubernetes-client');
+const BbPromise = require('bluebird');
+const helpers = require('../lib/helpers');
+const moment = require('moment');
+
+class KubelessLogs {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = this.serverless.getProvider('google');
+    this.commands = {
+      logs: {
+        usage: 'Output the logs of a deployed function',
+        lifecycleEvents: [
+          'logs',
+        ],
+        options: {
+          count: {
+            usage: 'Number of lines to print',
+            shortcut: 'n',
+          },
+        },
+      },
+    };
+    this.hooks = {
+      'logs:logs': () => BbPromise.bind(this)
+        .then(this.validate)
+        .then(this.logsFunction)
+        .then(() => {
+          if (this.options.tail) {
+            let m = moment.utc().valueOf();
+            setInterval(() => {
+              this.printLogs({
+                startTime: m,
+                count: null,
+              });
+              m = moment.utc().valueOf();
+            }, this.options.interval || 1000);
+          }
+        }),
+    };
+  }
+
+  validate() {
+    helpers.validateEnv();
+    const unsupportedOptions = ['stage', 'region'];
+    helpers.warnUnsupportedOptions(
+      unsupportedOptions,
+      this.options,
+      this.serverless.cli.log.bind(this.serverless.cli)
+    );
+    return BbPromise.resolve();
+  }
+
+  filterLogs(logs, options) {
+    const opts = _.defaults({}, options, {
+      startTime: null,
+      count: null,
+      filter: null,
+    });
+    let logEntries = _.compact(logs.split('\n'));
+    if (opts.count) {
+      logEntries = logEntries.slice(logEntries.length - opts.count);
+    }
+    if (opts.filter) {
+      logEntries = _.filter(logEntries, entry => !!entry.match(opts.filter));
+    }
+    if (opts.startTime) {
+      const since = !!opts.startTime.toString().match(/(?:m|h|d)/);
```

Comment:
is like doing `Boolean(var)`, with one `!` you are transforming the var into a boolean but negating it, with `!!` you restore the falsy/truthy value